net/ddns-scripts/files/usr/lib/ddns/dynamic_dns_updater.sh

```diff
@@ -397,7 +398,10 @@ while : ; do
 
 	# now we wait for check interval before testing if update was recognized
 	[ $DRY_RUN -eq 0 ] && {
-		write_log 7 "Waiting $CHECK_SECONDS seconds (Check Interval)"
+		get_uptime NOW_TIME
+		echo $(($NOW_TIME + $CHECK_SECONDS)) > $CHKFILE   # save the next scheduled check time
+		NEXT_CHECK_TIME=$( date -d @$(( $(date +%s) + $CHECK_SECONDS )) +"$ddns_dateformat" )
```

Comment:
I know what you're doing there, but can we write it in a way that's a bit easier to understand.
Outsource it to a subfunction? e.g. get_next_check_time().